### PR TITLE
87622 nb disability benefits kickoff failure mailers on lighthouse polled document failed

### PR DESCRIPTION
**Pull request by NB28VT**
## Summary

- *This work is behind a feature toggle (flipper): YES

Yes, adds a Flipper, `disability_compensation_email_veteran_on_polled_lighthouse_doc_failure` for toggling this behavior.

- *(Summarize the changes that have been made to the platform)*

TLDR Creates a new callback service for handling Form 526 evidence documents that have failed to process after we upload them to Lighthouse. Emails the veteran with instructions on a manual workaround to ensure the failure doesn't hold up their claim.

More detail:
We create `Lighthouse526DocumentUpload` records after we upload Form 526 evidence documents to Lighthouse so we can poll them later. We poll Lighthouse's `uploads/status` endpoint in the `Form526DocumentUploadPollingJob` to check on the status of the document. If there is a failure processing the document at Lighthouse, we mark the `Lighthouse526DocumentUpload` record to `failed`.

We already have emails to send the veteran if we have a problem uploading certain documents to Lighthouse in the first place, but they can fail later in the process when Lighthouse is processing them. We want to send the same emails in that situation, which is what this code does.

- *(Which team do you work for, does your team own the maintenance of this component?)*

Disability Benefits team 2 (aka Carbs); we own the maintenance of these components

- *(If introducing a flipper, what is the success criteria being targeted?)*

We will run this in staging for veteran evidence, form 0781 and form 0781a document failures, and ensure mailers get enqueued and sent out properly. Only then will we flip the toggle switch on. This release may be coordinated with other work and may not go out right away.

## Related issue(s)
- [Implementation ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/87622)
- [Testing ticket](https://github.com/orgs/department-of-veterans-affairs/projects/1263/views/7?pane=issue&itemId=85909918)

## Testing done

- [X] *New code is covered by unit tests*

## Acceptance criteria

- [X]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X]  No error nor warning in the console.
- [X]  Events are being sent to the appropriate logging solution
- [X]  Documentation has been updated (link to documentation)
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)

### CODE REVIEW

File: app/models/lighthouse526_document_upload.rb [hunk 1]
```diff
@@ -40,7 +42,7 @@ class Lighthouse526DocumentUpload < ApplicationRecord
     end
 
     event :fail do
-      transitions from: :pending, to: :failed, guard: %i[end_time_saved? error_message_saved?]
+      transitions from: :pending, to: :failed, guard: %i[end_time_saved? error_message_saved?], after: :handle_failure
```
Comment (tommasina-va) [inline app/models/lighthouse526_document_upload.rb:45]: This is a nifty callback!

**Review by tommasina-va — APPROVED** [review]
Looks great! Nice tests! :shipit:

**Comment by NB28VT** [comment]
Hmm the test failure looks like a timeout, seems to keep happening even after updating the branch, I think this is unrelated to my changes though

File: app/models/lighthouse526_document_upload.rb (hunk 1 above)
Comment (NB28VT) [reply app/models/lighthouse526_document_upload.rb:45]: Yeah, that AASM magic always enjoyed working with that gem!

**Review by RachalCassity — APPROVED** [review]

**Pull request merged**